a10_octavia/controller/worker/flows/a10_member_flows.py

```diff
@@ -129,31 +129,79 @@ def get_delete_member_flow(self):
-            requires=(constants.MEMBER, a10constants.VTHUNDER, constants.POOL)))
         delete_member_flow.add(database_tasks.DecrementMemberQuota(
             requires=constants.MEMBER))
-        if CONF.a10_global.network_type == 'vlan':
-            delete_member_flow.add(vthunder_tasks.DeleteInterfaceTagIfNotInUseForMember(
-                requires=[constants.MEMBER,
-                          a10constants.VTHUNDER]))
+        delete_member_flow.add(self.get_delete_member_vthunder_subflow())
         delete_member_flow.add(database_tasks.MarkPoolActiveInDB(
             requires=constants.POOL))
         delete_member_flow.add(database_tasks.
                                MarkLBAndListenersActiveInDB(
                                    requires=[constants.LOADBALANCER,
                                              constants.LISTENERS]))
-        delete_member_flow.add(a10_database_tasks.GetVRIDForProjectMember(
+        delete_member_flow.add(self.get_delete_member_vrid_subflow())
+        delete_member_flow.add(vthunder_tasks.WriteMemory(
+            requires=a10constants.VTHUNDER))
+        return delete_member_flow
+
+    def get_delete_member_vthunder_subflow(self):
+        delete_member_vthunder_subflow = linear_flow.Flow(
+            a10constants.DELETE_MEMBER_VTHUNDER_SUBFLOW)
+        delete_member_vthunder_subflow.add(server_tasks.MemberDelete(
+            requires=(constants.MEMBER, a10constants.VTHUNDER, constants.POOL)))
+        if CONF.a10_global.network_type == 'vlan':
+            delete_member_vthunder_subflow.add(vthunder_tasks.DeleteInterfaceTagIfNotInUseForMember(
+                requires=[constants.MEMBER, a10constants.VTHUNDER]))
+        return delete_member_vthunder_subflow
+
+    def get_delete_member_vthunder_internal_subflow(self, member_id):
+        delete_member_thunder_subflow = linear_flow.Flow(
+            a10constants.DELETE_MEMBER_VTHUNDER_SUBFLOW)
+        delete_member_thunder_subflow.add(vthunder_tasks.SetupDeviceNetworkMap(
+            name='setup_device_network_map_' + member_id,
+            requires=a10constants.VTHUNDER,
+            provides=a10constants.VTHUNDER))
+        delete_member_thunder_subflow.add(server_tasks.MemberDelete(
+            name='delete_thunder_member_' + member_id,
+            requires=(constants.MEMBER, a10constants.VTHUNDER, constants.POOL),
+            rebind={constants.MEMBER: member_id}))
+        if CONF.a10_global.network_type == 'vlan':
+            delete_member_thunder_subflow.add(vthunder_tasks.DeleteInterfaceTagIfNotInUseForMember(
+                name='delete_unused_interface_tag_in_member_' + member_id,
+                requires=[constants.MEMBER, a10constants.VTHUNDER],
+                rebind={constants.MEMBER: member_id}))
+
+        return delete_member_thunder_subflow
+
+    def get_delete_member_vrid_subflow(self):
+        delete_member_vrid_subflow = linear_flow.Flow(
+            a10constants.DELETE_MEMBER_VRID_SUBFLOW)
```

Comment:
`get_delete_member_vrid_subflow` and `a10constants.DELETE_MEMBER_VRID_SUBFLOW` shouldn't have the same subflow name.